gradio/components/audio.py

```diff
@@ -86,6 +87,7 @@ def __init__(
             elem_id: An optional string that is assigned as the id of this component in the HTML DOM. Can be used for targeting CSS styles.
             elem_classes: An optional list of strings that are assigned as the classes of this component in the HTML DOM. Can be used for targeting CSS styles.
             format: The file format to save audio files. Either 'wav' or 'mp3'. wav files are lossless but will tend to be larger files. mp3 files tend to be smaller. Default is wav. Applies both when this component is used as an input (when `type` is "format") and when this component is used as an output.
+            shareable: If True, will allow user to share generation on Hugging Face Spaces Discussions.
```

Comment:
```suggestion
            shareable: If True, will show a share icon in the corner of the component that allows user to share outputs to Hugging Face Spaces Discussions. If False, icon does not appear. If set to None (default behavior), then the icon appears if this Gradio app is launched on Spaces, but not otherwise.
```

Reply:
I think we should rename this to `show_share_button` because it is more consistent with the `show_copy_button` parameter which behaves similarly

Reply:
Done above, except:

> shareable=True does not work if you are not on spaces

if a user is going out of their way by setting show_share_button=True outside of a Space, it is presumably to just see what the button will look like on Spaces. I don't think it's necessary to handle what to do on click yet, now just has an Error modal, in the future when we add custom share listeners we can handle better.